processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -988,17 +992,27 @@ public QueryableIndex load(File inDir) throws IOException
        * Index.drd should consist of the segment version, the columns and dimensions of the segment as generic
        * indexes, the interval start and end millis as longs (in 16 bytes), and a bitmap index type.
        */
+
+      final GenericIndexed<String> metas = GenericIndexed.read(indexBuffer, GenericIndexed.STRING_STRATEGY);
       final GenericIndexed<String> cols = GenericIndexed.read(indexBuffer, GenericIndexed.STRING_STRATEGY);
       final GenericIndexed<String> dims = GenericIndexed.read(indexBuffer, GenericIndexed.STRING_STRATEGY);
+
       final Interval dataInterval = new Interval(indexBuffer.getLong(), indexBuffer.getLong());
       final BitmapSerdeFactory segmentBitmapSerdeFactory;
       /**
+       * meta data is indexed the same way cols, dims had
+       * */
+      Object commitMetaData = null;
+      for (String meta : metas) {
+      	commitMetaData = (Object) meta;
```

Comment:
is there a reason for storing metadata as GenericIndexed when it seems there just one?